helm-charts/splunk-otel-collector/templates/revert-patch-log-dirs-hook.yaml

```diff
@@ -0,0 +1,82 @@
+{{- if or (and (.Values.fluentd.securityContext.runAsUser) (.Values.fluentd.securityContext.runAsGroup)) (and (.Values.agent.securityContext.runAsUser) (.Values.agent.securityContext.runAsGroup)) }}
+apiVersion: v1
+kind: Pod
+metadata:
+  name: {{ template "splunk-otel-collector.fullname" . }}-revert-patch-log-dir
+  labels:
+    {{- include "splunk-otel-collector.commonLabels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": post-delete
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+spec:
+  restartPolicy: Never
+  containers:
+  - name: revert-patch-log-dirs
+    image: {{ template "splunk-otel-collector.image.initPatchLogDirs" . }}
+    imagePullPolicy: {{ .Values.image.initPatchLogDirs.pullPolicy }}
+    securityContext:
+      runAsUser: 0
+    command: ['sh', '-c', '
+    setfacl --recursive --remove-all  {{ .Values.logsCollection.checkpointPath }};
+    {{- if ne .Values.fluentd.config.posFilePrefix "/var/log/splunk-fluentd" }}
+    setfacl --recursive --remove-all  {{ dir .Values.fluentd.config.posFilePrefix }};
+    {{- end }}
+    {{ if .Values.logsCollection.containers.enabled -}}
+    if [ -d "/var/lib/docker/containers" ];
+    then
+        setfacl --recursive --remove-all  /var/lib/docker/containers;
+    fi;
+    if [ -d "/var/log/crio/pods" ];
+    then
+        setfacl --recursive --remove-all  /var/log/crio/pods;
+    fi;
+    if [ -d "/var/log/pods" ];
+    then
+        setfacl --recursive --remove-all  /var/log/pods;
+    fi;
+    {{- end }}
+    {{- if .Values.logsCollection.journald.enabled }}
+    if [ -d "{{ .Values.logsCollection.journald.directory }}" ];
+    then
+        setfacl --recursive --remove-all d:m::rx,m::rx,d:g:{{ .Values.agent.securityContext.runAsGroup | default 20000 }}:rx,g:{{ .Values.agent.securityContext.runAsGroup | default 20000 }}:rx {{ .Values.logsCollection.journald.directory }};
```

Comment:
Why is the default 20000 and not the splunk-otel-collector uid? https://github.com/signalfx/splunk-otel-collector/blob/main/cmd/otelcol/Dockerfile#L14. This would prevent non-native Smart Agent receiver functionality.

Reply:
Here we are deleting ACLs set by the init container. Currently 20000 is the default for user in the init container. This can be changed if it's preventing any functionalities.